From 5240e5d9055f7ce342c65cf2c4539830a67a7c1a Mon Sep 17 00:00:00 2001
Message-Id: <5240e5d9055f7ce342c65cf2c4539830a67a7c1a.1349887753.git.minovotn@redhat.com>
In-Reply-To: <8366ff5a206b0f75edd17a4a78f7c389ebb81d49.1349887753.git.minovotn@redhat.com>
References: <8366ff5a206b0f75edd17a4a78f7c389ebb81d49.1349887753.git.minovotn@redhat.com>
From: Alon Levy <alevy@redhat.com>
Date: Sun, 7 Oct 2012 14:26:48 +0200
Subject: [PATCH 10/11] qxl: better cleanup for surface destroy

RH-Author: Alon Levy <alevy@redhat.com>
Message-id: <1349620009-26192-7-git-send-email-alevy@redhat.com>
Patchwork-id: 42774
O-Subject: [PATCH RHEL-6.4 v2 6/7] qxl: better cleanup for surface destroy
Bugzilla: 820136
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Uri Lublin <uril@redhat.com>

From: Uri Lublin <uril@redhat.com>

Add back a call to qxl_spice_destroy_surface_wait_complete() in qxl_spice_destroy_surface_wait(),
that was removed by commit c480bb7da465186b84d8427e068ef7502e47ffbf

It is needed to complete surface-removal cleanup, for non async.
For async, qxl_spice_destroy_surface_wait_complete is called upon operation completion.

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>

Upstream: commit 753b8b0d77ba1b343a35f9679cc777ea10a62bba

clean cherry pick.
---
 hw/qxl.c | 1 +
 1 file changed, 1 insertion(+)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/qxl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/qxl.c b/hw/qxl.c
index 239322f..05abe91 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -184,6 +184,7 @@ static void qxl_spice_destroy_surface_wait(PCIQXLDevice *qxl, uint32_t id,
         spice_qxl_destroy_surface_async(&qxl->ssd.qxl, id, (uint64_t)cookie);
     } else {
         qxl->ssd.worker->destroy_surface_wait(qxl->ssd.worker, id);
+        qxl_spice_destroy_surface_wait_complete(qxl, id);
     }
 }
 
-- 
1.7.11.4

